libs/wire-api/src/Wire/API/Routes/Internal/Brig.hs

```diff
@@ -664,6 +669,81 @@ type TeamsAPI =
         :> ReqBody '[Servant.JSON] (Multi.TeamStatus SearchVisibilityInboundConfig)
         :> Post '[Servant.JSON] ()
     )
+    :<|> InvitationByEmail
+    :<|> InvitationCode
+    :<|> SuspendTeam
+    :<|> UnSuspendTeam
+    :<|> TeamSize
+    :<|> TeamInvitations
+
+type InvitationByEmail =
+  Named
+    "get-invitation-by-email"
+    ( "teams"
+        :> "invitations"
+        :> "by-email"
+        :> QueryParam' [Required, Strict] "email" Email
+        :> Get '[Servant.JSON] Invitation
+    )
+
+type InvitationCode =
+  Named
+    "get-invitation-code"
+    ( "teams"
+        :> "invitation-code"
+        :> QueryParam' [Required, Strict] "team" TeamId
+        :> QueryParam' [Required, Strict] "invitation_id" InvitationId
+        :> Get '[Servant.JSON] FoundInvitationCode
+    )
+
+newtype FoundInvitationCode = FoundInvitationCode User.InvitationCode
+  deriving stock (Eq, Show, Generic)
+  -- TODO: is this correct?
+  deriving newtype (S.ToSchema)
+
+instance ToJSON FoundInvitationCode where
+  toJSON (FoundInvitationCode c) = Aeson.object ["code" Aeson..= c]
+
+type SuspendTeam =
+  Named
+    "suspend-team"
+    ( "teams"
+        :> Capture "tid" TeamId
+        :> "suspend"
+        :> Post
+             '[Servant.JSON]
+             NoContent
+    )
+
+type UnSuspendTeam =
```

Comment:
`UnsuspendTeam` makes more sense to me. The current spelling makes it look like a typical Haskell unwrap of newtypes.

Reply:
haha, you're right, I will change it :D